checkstyle/src/main/java/org/elasticsearch/checkstyle/SnippetLengthCheck.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.checkstyle;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.function.BiConsumer;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.puppycrawl.tools.checkstyle.api.AbstractFileSetCheck;
+import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
+import com.puppycrawl.tools.checkstyle.api.FileText;
+
+/**
+ * Checks the snippets included in the docs aren't too wide to fit on
+ * the page.
+ */
+public class SnippetLengthCheck extends AbstractFileSetCheck {
+    private static final Pattern START = Pattern.compile("^( *)//\\s*tag::(.+?)\\s*$", Pattern.MULTILINE);
+    private int max;
+
+    /**
+     * The maximum width that a snippet may have.
+     */
+    public void setMax(int max) {
+        this.max = max;
+    }
+
+    @Override
+    @SuppressForbidden(reason = "Checkstyle's API contain File")
+    protected void processFiltered(File file, FileText fileText) throws CheckstyleException {
+        checkFile((line, message) -> log(line, message), max, fileText.toLinesArray());
```

Comment:
Seems a bit unnecessarily to plum this log function which just maps 1:1 with the `log()` method of the parent class. Is this just so that these methods can be `static`? Is there any real benefit to that? Does checkstyle create multiple instances of this?

Reply:
It means I don't have to figure out anything about the way checkstyle works to write the test. I could probably rework it so the test is more checkstyle-native if you'd prefer.